ls1intum/Artemis
# `Lectures`: Fix overlay on attachment edit page for Safari

### Checklist
#### General
#### Client

### Motivation and Context
fixes #6195
When editing an attachment, the whole screen was blocked by an overlay. This behaviour was encountered on Safari, but not on Chrome and Firefox

### Description
This pull request addresses an issue related to the behavior of the ```position: relative``` CSS property when applied to a table row (```tr```) in HTML5/CSS3 specifications. Currently, Safari ignores the position attribute on table rows, while Chrome and Firefox do not. As the behavior is not specified, this inconsistency is not considered a browser bug, but rather a misuse of the position attribute on a table row to restrict the size of our overlay.

The bug was fixed by removing the reliance on ```position: relative``` for the overlay. As a result, the text can now be horizontally centered, but vertical centering is not supported. Despite this limitation, the overlay is displayed exclusively above the corresponding table row, and the elements within the table row remain disabled.

These changes ensure consistent behavior across different browsers while maintaining the intended functionality of the overlay.


### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Lecture
- Test at least on Safari, preferably on Safari, Firefox and Chrome

1. Navigate to a Lecture (Course Management)
2. Create an attachment
3. Click edit for that attachment and verify that the overlay is displayed correctly

### Review Progress

#### Code Review
#### Manual Tests

### Screenshots

#### Before
Taken from #6195
<img width="1200" alt="Lectures_Attachments_Edit" src="https://user-images.githubusercontent.com/16115423/216505895-d450dce2-0bd8-4e51-95fb-e61c3df7a4df.png">

#### After
<img width="420" alt="Bildschirmfoto 2023-07-05 um 14 21 39" src="https://github.com/ls1intum/Artemis/assets/47261058/de9584e5-689a-4b32-9454-3ae5badfc41c">
